backend/src/main/java/com/empire/ArtificialIntelligence.java

```diff
@@ -642,6 +643,49 @@ boolean feasible() {
 		}
 	}
 
+	/** An intent for Cardinals to travel to the holy city and inspire. */
+	private class CardinalInspire extends Intent {
+		public CardinalInspire() {
+		}
+
+		@Override
+		boolean feasible() {
+			Ideology r = Nation.getStateReligion(whoami, world);
+			return (r.religion == Religion.IRUHAN && r != Ideology.VESSEL_OF_FAITH);
+		}
+
+		@Override
+		double importance() {
+			return 1;
+		}
+
+		double valueOf(Character piece) {
+			return piece.hasTag(Character.Tag.CARDINAL) ? 1.0 : 0;
+		}
+
+		@Override
+		Map<String, String> generateOrders() {
+			HashMap<String, String> orders = new HashMap<>();
+			final int holycityId = world.getGeography().holycity;
+			for (Character c : allocatedPieces.characters) {
+				if (c.location == holycityId) {
+					orders.put("action_" + c.name.replace(" ", "_").replace("'", "_"), "Inspire the Faithful");
+				} else {
+					Map<Region, Integer> regionDistances = world.regions.get(holycityId).getRegionsByDistance(world);
+					int currentDistance = regionDistances.get(world.regions.get(c.location));
+					int destination = -1;
+					for (Region r : world.regions) {
+						if (regionDistances.get(r) == currentDistance - 1 && r.getNeighborsIds(world).contains(c.location)) {
+							orders.put("action_" + c.name.replace(" ", "_").replace("'", "_"), "Travel to " + r.name);
```

Comment:
I suggest getting there by "Hide in" actions, since having stronger spy cardinals in generally better for the nation than general/admiral/governor cardinals.

Reply:
Updated.